state/accountsDB.go

```diff
@@ -1283,15 +1284,15 @@ func (adb *AccountsDB) processSnapshotCompletion(
 	handleLoggingWhenError("error while putting active DB value into main storer", errPut)
 }
 
-func (adb *AccountsDB) syncMissingNodes(missingNodesChan chan []byte, errChan chan error, stats *snapshotStatistics, syncer AccountsDBSyncer) {
+func (adb *AccountsDB) syncMissingNodes(missingNodesChan chan []byte, errChan common.BufferedErrChan, stats *snapshotStatistics, syncer AccountsDBSyncer) {
 	defer stats.SyncFinished()
 
 	if check.IfNil(syncer) {
 		log.Error("can not sync missing nodes", "error", ErrNilTrieSyncer.Error())
 		for missingNode := range missingNodesChan {
 			log.Warn("could not sync node", "hash", missingNode)
 		}
-		errChan <- ErrNilTrieSyncer
+		errChan.WriteInChanNonBlocking(ErrNilTrieSyncer)
```

Comment:
all these blocking calls became non-blocking calls. I guess this is desirable.

Reply:
Yes, this was the bug that prompted this refactoring.